src/main/java/org/mule/extension/sftp/internal/proxy/HttpParser.java

```diff
@@ -0,0 +1,214 @@
+                                    String header, int from) {
+    /*
+     * int length = header.length(); boolean first = true; for (int start = from; start <= length; first = false) { // Now we have
+     * either a single token, which may end in zero or more // equal signs, or a comma-separated list of key=value pairs (with //
+     * optional legacy whitespace around the equals sign), where the // value can be either a token or a quoted string. start =
+     * skipWhiteSpace(header, start); int end = HttpSupport.scanToken(header, start); if (end == start) { // Nothing found. Either
+     * at end or on a comma. if (start < header.length() && header.charAt(start) == ',') { return start + 1; } return start; } int
+     * next = skipWhiteSpace(header, end); // Comma, or equals sign, or end of string if (next >= length || header.charAt(next) !=
+     * '=') { if (first) { // It must be a token challenge.setToken(header.substring(start, end)); if (next < length &&
+     * header.charAt(next) == ',') { next++; } return next; } // This token must be the name of the next authentication // scheme.
+     * return start; } int nextStart = skipWhiteSpace(header, next + 1); if (nextStart >= length) { if (next == end) { // '='
+     * immediately after the key, no value: key must be the // token, and the equals sign is part of the token
+     * challenge.setToken(header.substring(start, end + 1)); } else { // Key without value...
+     * challenge.addArgument(header.substring(start, end), null); } return nextStart; } if (nextStart == end + 1 &&
+     * header.charAt(nextStart) == '=') { // More than one equals sign: must be the single token. end = nextStart + 1; while (end
+     * < length && header.charAt(end) == '=') { end++; } challenge.setToken(header.substring(start, end)); end =
+     * skipWhiteSpace(header, end); if (end < length && header.charAt(end) == ',') { end++; } return end; } if
+     * (header.charAt(nextStart) == ',') { if (next == end) { // '=' immediately after the key, no value: key must be the //
+     * token, and the equals sign is part of the token challenge.setToken(header.substring(start, end + 1)); return nextStart + 1;
+     * } // Key without value... challenge.addArgument(header.substring(start, end), null); start = nextStart + 1; } else { if
+     * (header.charAt(nextStart) == '"') { int[] nextEnd = { nextStart + 1 }; String value = scanQuotedString(header, nextStart +
+     * 1, nextEnd); challenge.addArgument(header.substring(start, end), value); start = nextEnd[0]; } else { int nextEnd =
+     * HttpSupport.scanToken(header, nextStart); challenge.addArgument(header.substring(start, end), header.substring(nextStart,
+     * nextEnd)); start = nextEnd; } start = skipWhiteSpace(header, start); if (start < length && header.charAt(start) == ',') {
+     * start++; } } } return length;
+     */
+    return 0;
+  }
+
+  private static int skipWhiteSpace(String header, int i) {
```

Comment:
What's wrong with trim function? Or replace with a match for leading spaces maybe